### PR TITLE
feat: link added for screenshot in report [#49]

### PR DESCRIPTION
**Pull request by anjaliahlawat**

### CODE REVIEW

**Comment by h2grover** [comment]
@anjaliahlawat can you please post the screenshot of the html report.

**Review by h2grover — APPROVED** [review]

**Comment by anjaliahlawat** [comment]
> @anjaliahlawat can you please post the screenshot of the html report.

screenshot of the page displaying resulted hyperlink of screenshot in html-report?

**Comment by h2grover** [comment]
> > @anjaliahlawat can you please post the screenshot of the html report.
> 
> screenshot of the page displaying resulted hyperlink of screenshot in html-report?

Yes and also the screenshot which is opened in the browser with path in the address bar

**Comment by anjaliahlawat** [comment]
Html report screenshot -
image

screenshot -
image

**Review by h2grover — CHANGES_REQUESTED** [review]
Anjali, we have to attach screenshot in case of failure only. In this case it will be attached even when we call takescreenshot function

**Comment by anjaliahlawat** [comment]
> Anjali, we have to attach screenshot in case of failure only. In this case it will be attached even when we call takescreenshot function

Oh right. Then I should add this in afterspec func under if condition. But since I need full path for screenshot, I should define this path in env. What do you think?

**Review by h2grover — CHANGES_REQUESTED** [review]
Anjali instead of passing isFailed argument, why don't you pass fullPath?
If fullPath is passed then we ill use that path otherwise takescreenshot will create its own path.

**Comment by h2grover** [comment]
Default path in the screenshot function will be reports/html-report/images/screenshots/

File: src/lib/PageWrapper.ts [hunk 1]
```diff
@@ -74,14 +74,22 @@ export default class PageWrapper {
...
+  async takeScreenShot(scrPath?: string): Promise<void> {
     const name = uniqueId();
-    const absPath = resolve(".");
-    const relPath = "reports/html-report/images/screenshots/failed";
-    await createDir(relPath);
-    const fullPath = join(absPath, relPath, `/${name}.png`);
-    await this.page.screenshot({
-      path: fullPath,
-    });
+    let fullPath: string;
```
Comment (h2grover) [inline src/lib/PageWrapper.ts]: please remove the redundant code

Comment (h2grover) [reply src/lib/PageWrapper.ts]:
```
await this.page.screenshot({
        path: fullPath,
      });
```

can be used only once.

**Review by h2grover — CHANGES_REQUESTED** [review]

File: src/lib/PageWrapper.ts [hunk 2]
```diff
@@ -74,12 +74,17 @@ export default class PageWrapper {
...
-    const fullPath = join(absPath, relPath, `/${name}.png`);
+    let fullPath: string;
+    if (scrPath) {
+      await createDir(scrPath);
+      fullPath = join(resolve("."), scrPath, `/${name}.png`);
+    } else {
+      const defaultScrPath = "reports/html-report/images/screenshots";
+      await createDir(defaultScrPath);
+      fullPath = join(resolve("."), defaultScrPath, `/${name}.png`);
+    }
```
Comment (h2grover) [inline src/lib/PageWrapper.ts]: you can also do like this.
```
  async takeScreenShot(scrPath?: string): Promise<void> {
    const name = uniqueId();
    const screenshotDirPath = scrPath || "reports/html-report/images/screenshots";
    await createDir(screenshotDirPath);
    const screenshotFullPath = join(resolve("."), scrPath, `/${name}.png`);
    await this.page.screenshot({
      path: screenshotFullPath,
    });
  }
```

**Review by h2grover — CHANGES_REQUESTED** [review]
please try this and try to remove the redundant code after reading it twice.

**Comment by h2grover** [comment]
@anjaliahlawat have you tested the `takeScrenshot` function with both the possibilities ?

**Comment by anjaliahlawat** [comment]
> @anjaliahlawat have you tested the `takeScrenshot` function with both the possibilities ?

Yes. It was working in both the cases.

**Review by h2grover — APPROVED** [review]

**Pull request merged**